src/Symfony/Installer/Tests/IntegrationTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the Symfony Installer package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Installer\Tests;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Filesystem\Filesystem;
+
+class IntegrationTest extends \PHPUnit_Framework_TestCase
+{
+    private static $rootDir = __DIR__.'/../../../../';
+    private $fs;
+
+    public function setUp()
+    {
+        $this->fs = new Filesystem();
+
+        $this->fs->remove(self::$rootDir.'/symfony.phar');
+        $this->runCommand('./vendor/bin/box build');
+    }
+
+    public static function tearDownAfterClass()
+    {
+        $fs = new Filesystem();
+        $fs->remove(self::$rootDir.'/symfony.phar');
+    }
+
+    public function testDemoApplicationInstallation()
+    {
+        $projectDir = sprintf('%s/my_test_project', sys_get_temp_dir());
+        $this->fs->remove($projectDir);
+
+        $output = $this->runCommand(sprintf('./symfony.phar demo %s', $projectDir));
+        $this->assertContains('Downloading the Symfony Demo Application', $output);
+        $this->assertContains('Symfony Demo Application was successfully installed.', $output);
+
+        $output = $this->runCommand(sprintf('cd %s && php app/console --version', $projectDir));
+        $this->assertRegExp('/Symfony version 2\.\d+\.\d+ - app\/dev\/debug/', $output);
+    }
+
+    /**
+     * @dataProvider provideSymfonyInstallationData
+     */
+    public function testSymfonyInstallation($additionalArguments, $messageRegexp, $versionRegexp)
+    {
+        $projectDir = sprintf('%s/my_test_project', sys_get_temp_dir());
+        $this->fs->remove($projectDir);
+
+        $commonArguments = sprintf('new %s', $projectDir);
+        $output = $this->runCommand(sprintf('./symfony.phar %s %s', $commonArguments, $additionalArguments));
+        $this->assertContains('Downloading Symfony...', $output);
+        $this->assertRegExp($messageRegexp, $output);
+
+        $output = $this->runCommand(sprintf('cd %s && php app/console --version', $projectDir));
+        $this->assertRegExp($versionRegexp, $output);
+    }
+
+    /**
+     * Runs the given string as a command and returns the resulting output.
+     * The CWD is set to the root project directory to simplify command paths.
+     *
+     * @param string $command
+     *
+     * @return string
+     *
+     * @throws RuntimeException in case the command execution is not successful
+     */
+    private function runCommand($command)
+    {
+        $process = new Process($command);
+        $process->setWorkingDirectory(self::$rootDir);
+        $process->run();
+
+        if (!$process->isSuccessful()) {
+            throw new \RuntimeException($process->getErrorOutput());
```

Comment:
use the ProcessFailedException, to give a better exception message on failure (you can even use ``$process->mustRun()`` rather than ``$process->run()``)